### PR TITLE
[varLib] avar required value maps

### PR DESCRIPTION
**Pull request by jenskutilek**
This adds the required value maps describes in #1011. Before merging, we must decide about the question in the issue regarding the spec intent.

It also changes the behaviour regarding empty avar segment maps:

If segment maps are present for any axis, DirectWrite currently requires to have explicit maps for all axes, even if they just contain the default entries. As can be verified in Chrome Canary on Windows Creators Update, the font is not loaded if any axis has an empty avar entry.

### CODE REVIEW

File: Lib/fontTools/varLib/__init__.py [hunk 1]
```diff
@@ -92,11 +92,15 @@ def _add_fvar_avar(font, axes, instances):
...
-		interesting = True
-
-		items = sorted(axis.map.items())
-		keys   = [item[0] for item in items]
+			interesting |= False
+		else:
+			interesting = True
+		if not axis.map:
+			# Deduce a default segment map for the axis
+			items = [(v, v) for v in sorted(list(set((axis.minimum, axis.default, axis.maximum))))]
```
Comment (anthrotype) [inline Lib/fontTools/varLib/__init__.py]: `sorted` already returns a list (copy), no need to do `sorted(list(set(...)))`

Comment (jenskutilek) [inline Lib/fontTools/varLib/__init__.py]: Thanks!

**Comment by PeterConstable** [comment]
The DirectWrite behaviour mentioned -- requiring explicit maps for all axes -- is a known bug in DirectWrite (the OT spec does not require explicit maps for all axes) and should not influence any decisions.

**Comment by jenskutilek** [comment]
So I guess most of this PR can be disregarded. I made a new PR: #1014

File: Lib/fontTools/varLib/__init__.py [hunk 2]
```diff
@@ -92,10 +92,9 @@ def _add_fvar_avar(font, axes, instances):
 	for axis in axes.values():
 		curve = avar.segments[axis.tag] = {}
 		if not axis.map or all(k==v for k,v in axis.map.items()):
-			continue
-		interesting = True
-
-		items = sorted(axis.map.items())
+			interesting |= False
```
Comment (behdad) [inline Lib/fontTools/varLib/__init__.py]: This line doesn't make sense.

Comment (jenskutilek) [inline Lib/fontTools/varLib/__init__.py]: It was supposed to make sure segment entries are being added even if not different from the default mapping after the first neccessary segment map has been found. It was a workaround for the DirectWrite bug.

I see now it's buggy …

**Pull request closed without merging**